src/main/sarl/fr/utbm/info/vi51/worldswar/simulator/Simulator.sarl

```diff
@@ -196,4 +208,28 @@ agent Simulator {
 		}
 	}
 	
+	private def pauseSimulation() {
+		cancel(this.stepTimerTask)
+		for(l : listeners) {
+			l.simulationPaused();
+		}
+	}
+	
+	private def resumeSimulation() {
+		this.stepTimerTask = in(this.simulationSpeed.milliseconds) [ setStepTimerReady ]
+		for(l : listeners) {
+			l.simulationResumed();
+		}
+	}
+	
+	private def stepSimulation() {
+		if(this.environmentReady) {
+			this.environmentReady = false
+			this.stepTimerReady = false
+			emit(new SimulationStepEvent)
+			for(l as SimulatorListener : this.listeners) {
```

Comment:
hmm bah comme ça on voit directement sur quel type d'objet on itère. C'est vrai que ça cause peut être de moindres performances, mais c'est pas sûr..